idea/run_configs.go

```diff
@@ -0,0 +1,418 @@
+				ConfigName:        entry.Name,
+				InstanceType:      entry.InstanceType,
+				InstanceNumber:    entry.InstanceNumber,
+				PortOffset:        entry.PortOffset,
+				UseEnvFile:        config.EnvFile,
+				PasswordSecret:    staticPasswordSecret,
+				RootPasswordHash:  fmt.Sprintf("%x", sha256.Sum256([]byte(config.RootPassword))),
+				DataDir:           entry.DataDir,
+				IsLeaderNode:      entry.InstanceNumber == 1, // Make the first node the leader node
+				OpenSearchVersion: opensearchVersion,
+			}
+
+			if entry.EnvTemplate != nil {
+				if err := entry.EnvTemplate.Execute(&entry.RenderedEnvTemplate, data); err != nil {
+					return fmt.Errorf("couldn't render env-file template: %w", err)
+				}
+			}
+
+			data.Env = gotenv.Parse(bytes.NewReader(entry.RenderedEnvTemplate.Bytes()))
+
+			if err := entry.Template.Execute(&entry.RenderedTemplate, data); err != nil {
+				return fmt.Errorf("couldn't compile template: %w", err)
+			}
+
+			entries = append(entries, entry)
+		}
+	}
+
+	// Write all run configuration entries to the file system
+	for _, entry := range entries {
+		if err := writeEntryFiles(rcDir, config, entry); err != nil {
+			return err
+		}
+	}
+
+	// Write all compound run configurations
+	for name, cfg := range configData.CompoundConfigs {
+		compoundFilename := fmt.Sprintf("%scompound-%s%s", generatedFilePrefix, name, runConfigSuffix)
+		compoundFilepath := filepath.Join(config.Workdir, runConfigDir, compoundFilename)
+
+		if _, err := os.Stat(compoundFilepath); !os.IsNotExist(err) && !config.Force {
+			logger.Info("Skipping existing compound configuration: %s", filepath.Join(runConfigDir, compoundFilename))
+			continue
+		}
+
+		filteredEntries := lo.Filter(entries, func(entry RunConfigEntry, index int) bool {
+			return slices.Contains(cfg.InstanceTypes, entry.InstanceType)
+		})
+
+		toRun := make([]CompoundToRun, 0)
+
+		for _, entry := range filteredEntries {
+			var xrc XMLRunConfig
+			if err := xml.NewDecoder(bytes.NewReader(entry.RenderedTemplate.Bytes())).Decode(&xrc); err != nil {
+				return fmt.Errorf("couldn't parse rendered run configuration %q: %w", entry.Filename, err)
+			}
+
+			toRun = append(toRun, CompoundToRun{Name: entry.Name, Type: xrc.Configuration.Type})
+		}
+
+		tmpl, err := template.New(cfg.Name).Parse(compoundTemplate)
+		if err != nil {
+			return fmt.Errorf("couldn't parse compound template %q: %w", cfg.Name, err)
+		}
+
+		var buf bytes.Buffer
+		if err := tmpl.Execute(&buf, map[string]any{"Name": cfg.Name, "ToRun": toRun}); err != nil {
+			return err
+		}
+
+		if err := utils.AtomicallyWriteFile(compoundFilepath, buf.Bytes(), 0600); err != nil {
+			return fmt.Errorf("couldn't write compound file %q: %w", compoundFilename, err)
+		}
+
+		logger.Info("Created compound configuration: %s", filepath.Join(runConfigDir, compoundFilename))
+	}
+
+	return nil
+}
+
+// The GRAYLOG_DATANODE_OPENSEARCH_LOCATION setting needs the OpenSearch version. Let's try to find it
```

Comment:
`GRAYLOG_DATANODE_OPENSEARCH_LOCATION` can be set to a parent directory that contains different distributions. We wouldn't have to detect the version then.

```
GRAYLOG_DATANODE_OPENSEARCH_LOCATION=../graylog-project-repos/graylog2-server/data-node/target/opensearch
```
should work.

Reply:
Thanks! Fixed.